quickcomm/models.py

```diff
@@ -1,3 +1,125 @@
+import uuid
 from django.db import models
+from django.contrib.auth.models import User
+from django.core.validators import URLValidator
+
+# NOTE: The models in this file do not take into account how the site will
+# interact with other APIs. Ideally, we should be able to reuse the model
+# classes, but this logic is to be implemented later.
+
+# Type fields are left out as they can be added later and are redundant. Host
+# fields are kept as the if these classes are used to represent remote authors,
+# the host field is necessary. ID is also left out.
+
+# We used UUIDs for pkeys to be more secure.
+# TODO what are the constraints on fields being null?
+
+class Author(models.Model):
+    """An author is a person associated with a user account via a one-to-one
+    relationship."""
+
+    # Example code for a one-to-one relationship taken from
+    # https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html#onetoone
+    # Author: Vitor Freitas
+    # Site: Simple is Better Than Complex
+    # Taken on: Feb 6, 2023
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    host = models.URLField(validators=[URLValidator])
+    display_name = models.CharField(max_length=100)
+    github = models.URLField(blank=True, null=True, validators=[URLValidator])
+    profile_image = models.URLField(blank=True, null=True, validators=[URLValidator])
+
+    # TODO determine if admins are authors
+    is_admin = models.BooleanField(default=False)
+
+    def follows(self, author):
+        """Returns true if this author (self) follows the given author."""
+        return Follow.objects.filter(follower=self, following=author).exists()
+
+    def following(self, author):
+        """Returns true if this author (self) is followed by the given author."""
+        return Follow.objects.filter(follower=author, following=self).exists()
+
```

Comment:
Question for the whole team, do we want to create a 'true friend' method as well in our model here that just returns
if both of these are true? Since we may need to call that functionality in a few places

Reply:
I see there's a method that will tell if a given relationship is bidirectional, but do we want to be able to list this for a user?